File: reshape-the-matrix/reshape-the-matrix.js
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
    let res = new Array(r).fill(0).map(()=> new Array(c).fill(0));
   
    
  //  if(mat.length )
    
    let data = [];
    
    if(mat.length + mat[0].length )
    //m * n
    for(let i=0; i < mat.length; i++){
        for(let j =0; j < mat[0].length; j++){
            data.push(mat[i][j])
        }
    }
    
    
    for(let i=0; i < res.length; i++){
        for(let j=0; j < res[0].length; j++){
            let d = data.shift();
            if(d != undefined){
               res[i][j] = d
            }else{
                return mat
            }
            
        }
    }
    
    return data.length > 0 ? mat : res 
};

File: reshape-the-matrix/reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        R = len(mat)
        C = len(mat[0])
        
        if R * C != r * c :
            return mat
        
        res = [[0] * c for _ in range(r)]
        
        sr = 0
        sc =0
        for i in range(0,len(mat)):
            for j in range(0,len(mat[0])):
                
                if sc == c:
                    sr +=1
                    sc =0
                
                res[sr][sc] = mat[i][j]
                sc +=1
        return res
                
        
        